### ISSUE TITLE
wine server processes not ending with some applications through gamescope

### ISSUE THREAD

**Issue by DonKatsu** [ValveSoftware/gamescope#1396]
When certain applications launched with gamescope close, their wine server processes don't end resulting in having to `wineserver -k`.
If gamescope is not used with those, the wine server processes end as expected when the application closes.

The game I noticed this with was a Unity game, Brown Dust 2. Other Unity games can have this issue, but some that I've tried do not.
I can observe the issue with wine's notepad and winemine, so it shouldn't just be a problem with Unity.

Using `env -u WAYLAND_DISPLAY gamescope`, wine's processes end properly when closing notepad/winemine. But BD2's still don't.

The result of this issue is restarts/shutdowns taking 1m30s while waiting on the stop job for the wine processes.

Distro: CachyOS
Kernel: 6.9.6
DE: KDE Plasma 6.1.0
GPU: AMD RX 6700 XT
Gamescope version: 3.14.22

**Comment by HanabishiRecca** [comment]
> The result of this issue is restarts/shutdowns taking 1m30s while waiting on the stop job for the wine processes.

Btw, you can work around that by tweaking systemd's [`DefaultTimeoutStopSec=`](https://www.freedesktop.org/software/systemd/man/latest/systemd-system.conf.html#DefaultTimeoutStartSec=) parameter.

Or just press `Alt+SysRq+i` [SysRq hotkey](https://wiki.archlinux.org/title/Keyboard_shortcuts#Kernel_(SysRq)) to kill blocking processes right away.

**Comment by layercak3** [comment]
Does this occur when the wine application gracefully exits/you alt+f4, or only when you close the gamescope window (by compositor sending close() on the toplevel via keybinding or a server decoration)?

**Comment by HanabishiRecca** [comment]
In my case it's both.

**Comment by DonKatsu** [comment]
Yep, happens whether I quit game/alt+f4 or hit the close button on the Plasma titlebar.

**Comment by layercak3** [comment]
Ok, I can't reproduce this issue then, sorry. I just had a separate issue with my custom pressure-vessel/proton wrapper and umu that had similar symptoms, but the processes/explorer.exe did respond to TERM so it just wasn't exiting. Sounds like your wine processes are actually getting stuck if it requires being sent sigkill by systemd after the timeout.

**Comment by HanabishiRecca** [comment]
https://github.com/ValveSoftware/gamescope/issues/1482#issuecomment-2336652645
